layers.go

```diff
@@ -228,10 +267,11 @@ func (r *layerStore) Load() error {
 				}
 				names[name] = layers[n]
 			}
-			if pslice, ok := parents[layer.Parent]; ok {
-				parents[layer.Parent] = append(pslice, layers[n])
-			} else {
-				parents[layer.Parent] = []*Layer{layers[n]}
+			if layer.CompressedDigest != "" {
+				compressedsums[layer.CompressedDigest] = append(compressedsums[layer.CompressedDigest], layer.ID)
+			}
+			if layer.UncompressedDigest != "" {
+				uncompressedsums[layer.UncompressedDigest] = append(uncompressedsums[layer.UncompressedDigest], layer.ID)
```

Comment:
This suggests that several different objects representing an exactly identical layer can exist in the layer store. Is that so?  (I have no idea, it may well be, e.g. if the store only contained filesystems resulting from applying the layer, i.e. the result of applying the layer may be different depending on parents.)